### ISSUE TITLE
initramfs local script should wait for devices to come online

### ISSUE THREAD

**Issue by HanSooloo** [zfsonlinux/pkg-zfs#4]
When booting with the ZFS enabled initramfs, my HDD's on a Marvell chipset don't come online fast enough for legacy style mounting during mountall.conf upstart job.
Consequently, mountall complains about not being able to see the dataset on my zpool.

In order to work around / fix this issue, it's suggested to wait for udev to finish recognizing _all_ the HDD's while the local script is running inside the initramfs.  Below is the recommended solution:
1.  vi /usr/share/initramfs-tools/scripts
2.  (find where the local-premount scripts are called in 'mountroot()', and add the magic line with 'wait_for_udev'):
   ...
   ...
   
   ```
   [ "$quiet" != "y" ] && log_begin_msg "Running /scripts/local-premount"
       run_scripts /scripts/local-premount
   [ "$quiet" != "y" ] && log_end_msg
   
   # added to simulate the zfs script
   wait_for_udev
   ```
   
   ...
   ...

**Comment by HanSooloo** [comment]
Diff output for proposed patch.  

```
diff --git a/local b/local
index 1021b81..6a6fe26 100644
--- a/local
+++ b/local
@@ -76,6 +76,9 @@ mountroot()
        run_scripts /scripts/local-premount
        [ "$quiet" != "y" ] && log_end_msg

+        # added to simulate zfs script
+        wait_for_udev
+
        if [ "${readonly}" = "y" ] && \
           [ -z "$LOOP" ]; then
                roflag=-r
```

Will try to provide a proper local-premount script later.

**Comment by dajhorn** [comment]
This issue was likely fixed by commit 41af745c17449004b717223807262f54b898d172.

Please reopen this ticket if the problem persists.
